internal/kafka/test/integration/kafka_expiration_test.go

```diff
@@ -0,0 +1,513 @@
+						},
+						SupportedBillingModels: []config.KafkaBillingModel{
+							{
+								ID:               types.STANDARD.String(),
+								AMSResource:      ocm.RHOSAKResourceName,
+								AMSProduct:       string(ocm.RHOSAKTrialProduct),
+								AMSBillingModels: []string{string(amsv1.BillingModelStandard)},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		// register dummy organisations and users to the quota management list configuration
+		quotaManagementListConfig.QuotaList = quota_management.RegisteredUsersListConfiguration{
+			Organisations: quota_management.OrganisationList{
+				{
+					Id:      dummyOrgID,
+					AnyUser: true,
+					GrantedQuota: quota_management.QuotaList{
+						{
+							InstanceTypeID: types.STANDARD.String(),
+							KafkaBillingModels: quota_management.BillingModelList{
+								{
+									Id: bmNoExpirationID,
+								},
+								{
+									Id:             bmNotExpiredID,
+									ExpirationDate: &[]quota_management.ExpirationDate{quota_management.ExpirationDate(time.Now().AddDate(0, 0, 10))}[0],
+								},
+								{
+									Id:             bmExpiredWithoutGracePeriodID,
+									ExpirationDate: &[]quota_management.ExpirationDate{quota_management.ExpirationDate(time.Now().AddDate(0, 0, -1))}[0],
+								},
+								{
+									Id:             bmExpiredWithGracePeriodID,
+									ExpirationDate: &[]quota_management.ExpirationDate{quota_management.ExpirationDate(time.Now().AddDate(0, 0, -1))}[0],
+								},
+							},
+						},
+					},
+				},
+			},
+			ServiceAccounts: quota_management.AccountList{
+				{
+					Username: dummyUser1,
+					GrantedQuota: quota_management.QuotaList{
+						{
+							InstanceTypeID: types.STANDARD.String(),
+							KafkaBillingModels: quota_management.BillingModelList{
+								{
+									Id: bmNoExpirationID,
+								},
+								{
+									Id:             bmNotExpiredID,
+									ExpirationDate: &[]quota_management.ExpirationDate{quota_management.ExpirationDate(time.Now().AddDate(0, 0, 10))}[0],
+								},
+								{
+									Id:             bmExpiredWithoutGracePeriodID,
+									ExpirationDate: &[]quota_management.ExpirationDate{quota_management.ExpirationDate(time.Now().AddDate(0, 0, -1))}[0],
+								},
+								{
+									Id:             bmExpiredWithGracePeriodID,
+									ExpirationDate: &[]quota_management.ExpirationDate{quota_management.ExpirationDate(time.Now().AddDate(0, 0, -1))}[0],
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+	})
+	defer tearDown()
+
+	mockKasFleetshardSyncBuilder := kasfleetshardsync.NewMockKasFleetshardSyncBuilder(h, t)
+	mockKasfFleetshardSync := mockKasFleetshardSyncBuilder.Build()
+	mockKasfFleetshardSync.Start()
+	defer mockKasfFleetshardSync.Stop()
+
+	clusterID, getClusterErr := common.GetRunningOsdClusterID(h, t)
+	if getClusterErr != nil {
+		t.Fatalf("Failed to retrieve cluster details: %v", getClusterErr)
+	}
+	if clusterID == "" {
+		panic("No cluster found")
+	}
```

Comment:
Do we need a running OSD? Alternatively we can directly seed a cluster in the database.